{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5ddd5ee7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import operator\n",
    "import plotly.express as px"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3629a1e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Genome</th>\n",
       "      <th>Ga0485165_metabat1.097_sub</th>\n",
       "      <th>Ga0485163_maxbin.015_sub</th>\n",
       "      <th>Ga0485161_metabat1.109_sub</th>\n",
       "      <th>Ga0485163_metabat1.131</th>\n",
       "      <th>Ga0485165_metabat2_ours.031</th>\n",
       "      <th>Ga0485163_metabat1.023_sub</th>\n",
       "      <th>Ga0485171_metabat2_ours.194</th>\n",
       "      <th>Ga0485163_maxbin.004_sub</th>\n",
       "      <th>Ga0485162_metabat2_ours.116</th>\n",
       "      <th>Ga0485169_metabat2_ours.243_sub</th>\n",
       "      <th>...</th>\n",
       "      <th>Ga0485171_metabat2_ours.107_sub</th>\n",
       "      <th>Ga0485168_metabat2_ours.136_sub</th>\n",
       "      <th>Ga0485165_metabat2_ours.033</th>\n",
       "      <th>Ga0485169_metabat2_jgi.026_sub</th>\n",
       "      <th>Ga0485169_metabat2_ours.028</th>\n",
       "      <th>Ga0485157_maxbin.139_sub</th>\n",
       "      <th>Ga0485164_metabat2_jgi.001</th>\n",
       "      <th>Ga0485168_metabat2_ours.052_sub</th>\n",
       "      <th>Ga0485160_maxbin.035_sub</th>\n",
       "      <th>Ga0485159_metabat2_ours.140_sub</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ME_2020_07_24_15m_B.fastq Read Count</th>\n",
       "      <td>929</td>\n",
       "      <td>136</td>\n",
       "      <td>21252</td>\n",
       "      <td>78567</td>\n",
       "      <td>203159</td>\n",
       "      <td>1456</td>\n",
       "      <td>296</td>\n",
       "      <td>285</td>\n",
       "      <td>1456</td>\n",
       "      <td>15155</td>\n",
       "      <td>...</td>\n",
       "      <td>7411</td>\n",
       "      <td>870</td>\n",
       "      <td>935</td>\n",
       "      <td>195</td>\n",
       "      <td>10</td>\n",
       "      <td>8480</td>\n",
       "      <td>27255</td>\n",
       "      <td>1467</td>\n",
       "      <td>2862</td>\n",
       "      <td>398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_07_24_23_5M_B.fastq Read Count</th>\n",
       "      <td>12747</td>\n",
       "      <td>264</td>\n",
       "      <td>7253</td>\n",
       "      <td>11930</td>\n",
       "      <td>83501</td>\n",
       "      <td>585</td>\n",
       "      <td>49</td>\n",
       "      <td>80</td>\n",
       "      <td>4612</td>\n",
       "      <td>23301</td>\n",
       "      <td>...</td>\n",
       "      <td>2197</td>\n",
       "      <td>260</td>\n",
       "      <td>2628</td>\n",
       "      <td>905</td>\n",
       "      <td>8</td>\n",
       "      <td>3332</td>\n",
       "      <td>5619</td>\n",
       "      <td>848</td>\n",
       "      <td>596</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_07_24_5m_C.fastq Read Count</th>\n",
       "      <td>24</td>\n",
       "      <td>538</td>\n",
       "      <td>7615</td>\n",
       "      <td>201</td>\n",
       "      <td>50</td>\n",
       "      <td>8188</td>\n",
       "      <td>139</td>\n",
       "      <td>401</td>\n",
       "      <td>16</td>\n",
       "      <td>101</td>\n",
       "      <td>...</td>\n",
       "      <td>1141</td>\n",
       "      <td>3024</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>2125</td>\n",
       "      <td>199</td>\n",
       "      <td>1713</td>\n",
       "      <td>3232</td>\n",
       "      <td>6459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_08_05_10M_B.fastq Read Count</th>\n",
       "      <td>40</td>\n",
       "      <td>589</td>\n",
       "      <td>3019</td>\n",
       "      <td>305572</td>\n",
       "      <td>3168</td>\n",
       "      <td>5636</td>\n",
       "      <td>155</td>\n",
       "      <td>1471</td>\n",
       "      <td>49</td>\n",
       "      <td>639</td>\n",
       "      <td>...</td>\n",
       "      <td>1156</td>\n",
       "      <td>1896</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>12058</td>\n",
       "      <td>14124</td>\n",
       "      <td>946</td>\n",
       "      <td>8933</td>\n",
       "      <td>1112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_08_05_15M_B.fastq Read Count</th>\n",
       "      <td>416</td>\n",
       "      <td>199</td>\n",
       "      <td>15025</td>\n",
       "      <td>60601</td>\n",
       "      <td>381617</td>\n",
       "      <td>1039</td>\n",
       "      <td>127</td>\n",
       "      <td>289</td>\n",
       "      <td>1084</td>\n",
       "      <td>19170</td>\n",
       "      <td>...</td>\n",
       "      <td>6448</td>\n",
       "      <td>788</td>\n",
       "      <td>249</td>\n",
       "      <td>197</td>\n",
       "      <td>5</td>\n",
       "      <td>4535</td>\n",
       "      <td>37094</td>\n",
       "      <td>1194</td>\n",
       "      <td>2424</td>\n",
       "      <td>286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_08_05_23_5M_B.fastq Read Count</th>\n",
       "      <td>14680</td>\n",
       "      <td>224</td>\n",
       "      <td>3782</td>\n",
       "      <td>4554</td>\n",
       "      <td>63741</td>\n",
       "      <td>833</td>\n",
       "      <td>25</td>\n",
       "      <td>129</td>\n",
       "      <td>4184</td>\n",
       "      <td>16194</td>\n",
       "      <td>...</td>\n",
       "      <td>1129</td>\n",
       "      <td>340</td>\n",
       "      <td>2197</td>\n",
       "      <td>1135</td>\n",
       "      <td>13</td>\n",
       "      <td>1676</td>\n",
       "      <td>3541</td>\n",
       "      <td>460</td>\n",
       "      <td>894</td>\n",
       "      <td>287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_08_25_10M_B.fastq Read Count</th>\n",
       "      <td>5</td>\n",
       "      <td>961</td>\n",
       "      <td>1686</td>\n",
       "      <td>96021</td>\n",
       "      <td>466</td>\n",
       "      <td>18082</td>\n",
       "      <td>358</td>\n",
       "      <td>6800</td>\n",
       "      <td>13</td>\n",
       "      <td>184</td>\n",
       "      <td>...</td>\n",
       "      <td>695</td>\n",
       "      <td>1652</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>1708</td>\n",
       "      <td>618</td>\n",
       "      <td>1557</td>\n",
       "      <td>1093</td>\n",
       "      <td>392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_08_25_15M_B.fastq Read Count</th>\n",
       "      <td>2190</td>\n",
       "      <td>539</td>\n",
       "      <td>9722</td>\n",
       "      <td>110259</td>\n",
       "      <td>457059</td>\n",
       "      <td>2505</td>\n",
       "      <td>21</td>\n",
       "      <td>658</td>\n",
       "      <td>715</td>\n",
       "      <td>23537</td>\n",
       "      <td>...</td>\n",
       "      <td>1666</td>\n",
       "      <td>983</td>\n",
       "      <td>6343</td>\n",
       "      <td>718</td>\n",
       "      <td>88</td>\n",
       "      <td>3935</td>\n",
       "      <td>24189</td>\n",
       "      <td>1002</td>\n",
       "      <td>751</td>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_08_25_23_5M_B.fastq Read Count</th>\n",
       "      <td>14941</td>\n",
       "      <td>252</td>\n",
       "      <td>2107</td>\n",
       "      <td>13449</td>\n",
       "      <td>175395</td>\n",
       "      <td>1473</td>\n",
       "      <td>7</td>\n",
       "      <td>174</td>\n",
       "      <td>2621</td>\n",
       "      <td>35404</td>\n",
       "      <td>...</td>\n",
       "      <td>966</td>\n",
       "      <td>358</td>\n",
       "      <td>28963</td>\n",
       "      <td>4699</td>\n",
       "      <td>68</td>\n",
       "      <td>1961</td>\n",
       "      <td>3535</td>\n",
       "      <td>592</td>\n",
       "      <td>570</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_09_11_15M_B.fastq Read Count</th>\n",
       "      <td>4319</td>\n",
       "      <td>350</td>\n",
       "      <td>5168</td>\n",
       "      <td>47148</td>\n",
       "      <td>230966</td>\n",
       "      <td>3210</td>\n",
       "      <td>40</td>\n",
       "      <td>501</td>\n",
       "      <td>1691</td>\n",
       "      <td>14586</td>\n",
       "      <td>...</td>\n",
       "      <td>851</td>\n",
       "      <td>3326</td>\n",
       "      <td>5843</td>\n",
       "      <td>7329</td>\n",
       "      <td>482</td>\n",
       "      <td>2476</td>\n",
       "      <td>8663</td>\n",
       "      <td>399</td>\n",
       "      <td>308</td>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_09_11_23_5M_B.fastq Read Count</th>\n",
       "      <td>7175</td>\n",
       "      <td>98</td>\n",
       "      <td>1019</td>\n",
       "      <td>5533</td>\n",
       "      <td>141118</td>\n",
       "      <td>2181</td>\n",
       "      <td>10</td>\n",
       "      <td>276</td>\n",
       "      <td>2229</td>\n",
       "      <td>24369</td>\n",
       "      <td>...</td>\n",
       "      <td>440</td>\n",
       "      <td>1111</td>\n",
       "      <td>9357</td>\n",
       "      <td>9902</td>\n",
       "      <td>574</td>\n",
       "      <td>845</td>\n",
       "      <td>1549</td>\n",
       "      <td>191</td>\n",
       "      <td>202</td>\n",
       "      <td>127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_10_08_15M_B.fastq Read Count</th>\n",
       "      <td>20</td>\n",
       "      <td>646</td>\n",
       "      <td>2913</td>\n",
       "      <td>621</td>\n",
       "      <td>411</td>\n",
       "      <td>4754</td>\n",
       "      <td>1116</td>\n",
       "      <td>1481</td>\n",
       "      <td>37</td>\n",
       "      <td>303</td>\n",
       "      <td>...</td>\n",
       "      <td>2618</td>\n",
       "      <td>25545</td>\n",
       "      <td>26</td>\n",
       "      <td>140</td>\n",
       "      <td>46</td>\n",
       "      <td>1178</td>\n",
       "      <td>240</td>\n",
       "      <td>3733</td>\n",
       "      <td>478</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_10_08_23_5M_B.fastq Read Count</th>\n",
       "      <td>3323</td>\n",
       "      <td>95</td>\n",
       "      <td>1246</td>\n",
       "      <td>13190</td>\n",
       "      <td>47527</td>\n",
       "      <td>1979</td>\n",
       "      <td>20</td>\n",
       "      <td>262</td>\n",
       "      <td>3007</td>\n",
       "      <td>29573</td>\n",
       "      <td>...</td>\n",
       "      <td>431</td>\n",
       "      <td>2199</td>\n",
       "      <td>3562</td>\n",
       "      <td>18528</td>\n",
       "      <td>981</td>\n",
       "      <td>849</td>\n",
       "      <td>1226</td>\n",
       "      <td>244</td>\n",
       "      <td>172</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_10_19_15M_B.fastq Read Count</th>\n",
       "      <td>7</td>\n",
       "      <td>443</td>\n",
       "      <td>4111</td>\n",
       "      <td>186</td>\n",
       "      <td>178</td>\n",
       "      <td>7391</td>\n",
       "      <td>3022</td>\n",
       "      <td>1648</td>\n",
       "      <td>53</td>\n",
       "      <td>209</td>\n",
       "      <td>...</td>\n",
       "      <td>3988</td>\n",
       "      <td>12931</td>\n",
       "      <td>19</td>\n",
       "      <td>54</td>\n",
       "      <td>29</td>\n",
       "      <td>1053</td>\n",
       "      <td>66</td>\n",
       "      <td>925</td>\n",
       "      <td>106</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_10_19_23_5M_B.fastq Read Count</th>\n",
       "      <td>173</td>\n",
       "      <td>327</td>\n",
       "      <td>4148</td>\n",
       "      <td>1804</td>\n",
       "      <td>3065</td>\n",
       "      <td>5073</td>\n",
       "      <td>1964</td>\n",
       "      <td>1159</td>\n",
       "      <td>321</td>\n",
       "      <td>3105</td>\n",
       "      <td>...</td>\n",
       "      <td>3745</td>\n",
       "      <td>10011</td>\n",
       "      <td>121</td>\n",
       "      <td>1070</td>\n",
       "      <td>68</td>\n",
       "      <td>1385</td>\n",
       "      <td>84</td>\n",
       "      <td>714</td>\n",
       "      <td>81</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ME_2020_10_19_5M_B.fastq Read Count</th>\n",
       "      <td>10</td>\n",
       "      <td>379</td>\n",
       "      <td>4237</td>\n",
       "      <td>198</td>\n",
       "      <td>112</td>\n",
       "      <td>4952</td>\n",
       "      <td>2680</td>\n",
       "      <td>1532</td>\n",
       "      <td>56</td>\n",
       "      <td>160</td>\n",
       "      <td>...</td>\n",
       "      <td>4181</td>\n",
       "      <td>13026</td>\n",
       "      <td>5</td>\n",
       "      <td>66</td>\n",
       "      <td>28</td>\n",
       "      <td>1125</td>\n",
       "      <td>57</td>\n",
       "      <td>937</td>\n",
       "      <td>87</td>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16 rows × 431 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Genome                                 Ga0485165_metabat1.097_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                          929   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                      12747   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                            24   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                           40   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                          416   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                      14680   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                            5   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                         2190   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                      14941   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                         4319   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                       7175   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                           20   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                       3323   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                            7   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                        173   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                            10   \n",
       "\n",
       "Genome                                 Ga0485163_maxbin.015_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                        136   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                      264   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                         538   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                        589   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                        199   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                      224   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                        961   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                        539   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                      252   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                        350   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                       98   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                        646   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                       95   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                        443   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                      327   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                         379   \n",
       "\n",
       "Genome                                 Ga0485161_metabat1.109_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                        21252   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                       7253   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                          7615   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                         3019   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                        15025   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                       3782   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                         1686   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                         9722   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                       2107   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                         5168   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                       1019   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                         2913   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                       1246   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                         4111   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                       4148   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                          4237   \n",
       "\n",
       "Genome                                 Ga0485163_metabat1.131  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                    78567   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                  11930   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                       201   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                   305572   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                    60601   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                   4554   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                    96021   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                   110259   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                  13449   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                    47148   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                   5533   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                      621   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                  13190   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                      186   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                   1804   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                       198   \n",
       "\n",
       "Genome                                 Ga0485165_metabat2_ours.031  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                        203159   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                       83501   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                             50   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                          3168   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                        381617   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                       63741   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                           466   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                        457059   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                      175395   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                        230966   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                      141118   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                           411   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                       47527   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                           178   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                        3065   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                            112   \n",
       "\n",
       "Genome                                 Ga0485163_metabat1.023_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                         1456   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                        585   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                          8188   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                         5636   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                         1039   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                        833   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                        18082   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                         2505   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                       1473   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                         3210   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                       2181   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                         4754   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                       1979   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                         7391   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                       5073   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                          4952   \n",
       "\n",
       "Genome                                 Ga0485171_metabat2_ours.194  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                           296   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                          49   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                            139   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                           155   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                           127   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                          25   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                           358   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                            21   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                           7   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                            40   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                          10   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                          1116   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                          20   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                          3022   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                        1964   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                           2680   \n",
       "\n",
       "Genome                                 Ga0485163_maxbin.004_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                        285   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                       80   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                         401   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                       1471   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                        289   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                      129   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                       6800   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                        658   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                      174   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                        501   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                      276   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                       1481   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                      262   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                       1648   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                     1159   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                        1532   \n",
       "\n",
       "Genome                                 Ga0485162_metabat2_ours.116  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                          1456   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                        4612   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                             16   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                            49   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                          1084   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                        4184   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                            13   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                           715   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                        2621   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                          1691   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                        2229   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                            37   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                        3007   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                            53   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                         321   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                             56   \n",
       "\n",
       "Genome                                 Ga0485169_metabat2_ours.243_sub  ...  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                             15155  ...   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                           23301  ...   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                                101  ...   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                               639  ...   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                             19170  ...   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                           16194  ...   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                               184  ...   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                             23537  ...   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                           35404  ...   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                             14586  ...   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                           24369  ...   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                               303  ...   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                           29573  ...   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                               209  ...   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                            3105  ...   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                                160  ...   \n",
       "\n",
       "Genome                                 Ga0485171_metabat2_ours.107_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                              7411   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                            2197   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                               1141   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                              1156   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                              6448   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                            1129   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                               695   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                              1666   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                             966   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                               851   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                             440   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                              2618   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                             431   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                              3988   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                            3745   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                               4181   \n",
       "\n",
       "Genome                                 Ga0485168_metabat2_ours.136_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                               870   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                             260   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                               3024   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                              1896   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                               788   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                             340   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                              1652   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                               983   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                             358   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                              3326   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                            1111   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                             25545   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                            2199   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                             12931   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                           10011   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                              13026   \n",
       "\n",
       "Genome                                 Ga0485165_metabat2_ours.033  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                           935   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                        2628   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                             17   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                            13   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                           249   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                        2197   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                            17   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                          6343   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                       28963   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                          5843   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                        9357   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                            26   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                        3562   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                            19   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                         121   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                              5   \n",
       "\n",
       "Genome                                 Ga0485169_metabat2_jgi.026_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                              195   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                            905   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                                 2   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                               10   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                              197   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                           1135   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                                2   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                              718   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                           4699   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                             7329   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                           9902   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                              140   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                          18528   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                               54   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                           1070   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                                66   \n",
       "\n",
       "Genome                                 Ga0485169_metabat2_ours.028  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                            10   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                           8   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                             14   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                             6   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                             5   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                          13   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                             8   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                            88   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                          68   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                           482   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                         574   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                            46   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                         981   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                            29   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                          68   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                             28   \n",
       "\n",
       "Genome                                 Ga0485157_maxbin.139_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                       8480   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                     3332   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                        2125   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                      12058   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                       4535   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                     1676   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                       1708   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                       3935   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                     1961   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                       2476   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                      845   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                       1178   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                      849   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                       1053   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                     1385   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                        1125   \n",
       "\n",
       "Genome                                 Ga0485164_metabat2_jgi.001  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                        27255   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                       5619   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                           199   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                        14124   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                        37094   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                       3541   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                          618   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                        24189   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                       3535   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                         8663   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                       1549   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                          240   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                       1226   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                           66   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                         84   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                            57   \n",
       "\n",
       "Genome                                 Ga0485168_metabat2_ours.052_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                              1467   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                             848   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                               1713   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                               946   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                              1194   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                             460   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                              1557   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                              1002   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                             592   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                               399   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                             191   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                              3733   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                             244   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                               925   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                             714   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                                937   \n",
       "\n",
       "Genome                                 Ga0485160_maxbin.035_sub  \\\n",
       "ME_2020_07_24_15m_B.fastq Read Count                       2862   \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                      596   \n",
       "ME_2020_07_24_5m_C.fastq Read Count                        3232   \n",
       "ME_2020_08_05_10M_B.fastq Read Count                       8933   \n",
       "ME_2020_08_05_15M_B.fastq Read Count                       2424   \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                      894   \n",
       "ME_2020_08_25_10M_B.fastq Read Count                       1093   \n",
       "ME_2020_08_25_15M_B.fastq Read Count                        751   \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                      570   \n",
       "ME_2020_09_11_15M_B.fastq Read Count                        308   \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                      202   \n",
       "ME_2020_10_08_15M_B.fastq Read Count                        478   \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                      172   \n",
       "ME_2020_10_19_15M_B.fastq Read Count                        106   \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                       81   \n",
       "ME_2020_10_19_5M_B.fastq Read Count                          87   \n",
       "\n",
       "Genome                                 Ga0485159_metabat2_ours.140_sub  \n",
       "ME_2020_07_24_15m_B.fastq Read Count                               398  \n",
       "ME_2020_07_24_23_5M_B.fastq Read Count                             108  \n",
       "ME_2020_07_24_5m_C.fastq Read Count                               6459  \n",
       "ME_2020_08_05_10M_B.fastq Read Count                              1112  \n",
       "ME_2020_08_05_15M_B.fastq Read Count                               286  \n",
       "ME_2020_08_05_23_5M_B.fastq Read Count                             287  \n",
       "ME_2020_08_25_10M_B.fastq Read Count                               392  \n",
       "ME_2020_08_25_15M_B.fastq Read Count                               311  \n",
       "ME_2020_08_25_23_5M_B.fastq Read Count                             126  \n",
       "ME_2020_09_11_15M_B.fastq Read Count                               217  \n",
       "ME_2020_09_11_23_5M_B.fastq Read Count                             127  \n",
       "ME_2020_10_08_15M_B.fastq Read Count                                90  \n",
       "ME_2020_10_08_23_5M_B.fastq Read Count                             126  \n",
       "ME_2020_10_19_15M_B.fastq Read Count                               139  \n",
       "ME_2020_10_19_23_5M_B.fastq Read Count                              73  \n",
       "ME_2020_10_19_5M_B.fastq Read Count                                 92  \n",
       "\n",
       "[16 rows x 431 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv(\"Fw_ lake mendota data/coverm_431_MAGS_metatransciptome_count.tsv\", sep='\\t')\n",
    "data=data.T\n",
    "data.columns = data.iloc[0]\n",
    "\n",
    "# drop the first row\n",
    "data= data.iloc[1:]\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5cdb4231",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_csv(\"metatransciptome_count.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e402b3c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#update the daa with all the environmental features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "908d7385",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(                       Genome       Domain            Phylum            Class  \\\n",
       " 0  Ga0485163_metabat1.110_sub  d__Bacteria  p__Chloroflexota  c__Anaerolineae   \n",
       " 1        Ga0485159_maxbin.015  d__Bacteria   p__Bacteroidota   c__Bacteroidia   \n",
       " 2  Ga0485168_metabat1.080_sub  d__Bacteria   p__Bacteroidota   c__Bacteroidia   \n",
       " 3    Ga0485168_maxbin.166_sub  d__Bacteria   p__Bacteroidota   c__Bacteroidia   \n",
       " 4    Ga0485170_maxbin.116_sub  d__Bacteria   p__Bacteroidota   c__Bacteroidia   \n",
       " \n",
       "                  Order              Family      Genus                Species  \n",
       " 0     o__Caldilineales   f__Caldilineaceae    g__CFX5                    s__  \n",
       " 1   o__Chitinophagales         f__UBA10324  g__BJGO01  s__BJGO01 sp014190535  \n",
       " 2           o__AKYH767  f__2-12-FULL-35-15        g__                    s__  \n",
       " 3  o__Flavobacteriales  f__Schleiferiaceae  g__TMED14                    s__  \n",
       " 4  o__Flavobacteriales  f__Schleiferiaceae  g__TMED14                    s__  ,\n",
       "                                Unnamed: 0  Ga0485165_metabat1.097_sub  \\\n",
       " 0    ME_2020_07_24_15m_B.fastq Read Count                         929   \n",
       " 1  ME_2020_07_24_23_5M_B.fastq Read Count                       12747   \n",
       " 2     ME_2020_07_24_5m_C.fastq Read Count                          24   \n",
       " 3    ME_2020_08_05_10M_B.fastq Read Count                          40   \n",
       " 4    ME_2020_08_05_15M_B.fastq Read Count                         416   \n",
       " \n",
       "    Ga0485163_maxbin.015_sub  Ga0485161_metabat1.109_sub  \\\n",
       " 0                       136                       21252   \n",
       " 1                       264                        7253   \n",
       " 2                       538                        7615   \n",
       " 3                       589                        3019   \n",
       " 4                       199                       15025   \n",
       " \n",
       "    Ga0485163_metabat1.131  Ga0485165_metabat2_ours.031  \\\n",
       " 0                   78567                       203159   \n",
       " 1                   11930                        83501   \n",
       " 2                     201                           50   \n",
       " 3                  305572                         3168   \n",
       " 4                   60601                       381617   \n",
       " \n",
       "    Ga0485163_metabat1.023_sub  Ga0485171_metabat2_ours.194  \\\n",
       " 0                        1456                          296   \n",
       " 1                         585                           49   \n",
       " 2                        8188                          139   \n",
       " 3                        5636                          155   \n",
       " 4                        1039                          127   \n",
       " \n",
       "    Ga0485163_maxbin.004_sub  Ga0485162_metabat2_ours.116  ...  depth  \\\n",
       " 0                       285                         1456  ...   15.0   \n",
       " 1                        80                         4612  ...   23.5   \n",
       " 2                       401                           16  ...    5.0   \n",
       " 3                      1471                           49  ...   10.0   \n",
       " 4                       289                         1084  ...   15.0   \n",
       " \n",
       "    wtemp_in_celsius  DO_sat_perc  DO_mg_L  specific_conductivity    PH  \\\n",
       " 0             12.03         -1.7    -0.18                 581.66  7.37   \n",
       " 1             10.31         -1.5    -0.16                 600.14  7.17   \n",
       " 2             25.05        109.1     9.00                 492.70  8.62   \n",
       " 3             18.50         -0.7    -0.07                 570.32  7.47   \n",
       " 4             11.76         -1.1    -0.12                 582.32  7.41   \n",
       " \n",
       "    chlorophyll_RFU  phycocyanin_RFU  Dissolved_organic_matter_RFU  \\\n",
       " 0             0.14             0.53                         12.18   \n",
       " 1             0.35             0.88                         12.65   \n",
       " 2             0.56             1.70                          9.70   \n",
       " 3             0.33             0.87                         11.63   \n",
       " 4             0.22             0.67                         12.18   \n",
       " \n",
       "    Turbidity_RFU  \n",
       " 0           0.59  \n",
       " 1           1.71  \n",
       " 2           4.99  \n",
       " 3           0.86  \n",
       " 4           0.72  \n",
       " \n",
       " [5 rows x 442 columns])"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load the taxonomy file again (repeating due to re-upload)\n",
    "taxonomy_file_path = 'Fw_ lake mendota data/MAG_taxonomy.tsv'\n",
    "taxonomy_df = pd.read_csv(taxonomy_file_path, sep='\\t')\n",
    "\n",
    "# Load the T-count file\n",
    "t_count_file_path = 'metatransciptome_count.csv'\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "\n",
    "# Display the first few rows of each dataframe to ensure correct loading\n",
    "(taxonomy_df.head(), t_count_df.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d208e06",
   "metadata": {},
   "source": [
    "# top 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "3a8de271",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Ga0485165_metabat1.097_sub</th>\n",
       "      <th>Ga0485163_maxbin.015_sub</th>\n",
       "      <th>Ga0485161_metabat1.109_sub</th>\n",
       "      <th>Ga0485163_metabat1.131</th>\n",
       "      <th>Ga0485165_metabat2_ours.031</th>\n",
       "      <th>Ga0485163_metabat1.023_sub</th>\n",
       "      <th>Ga0485171_metabat2_ours.194</th>\n",
       "      <th>Ga0485163_maxbin.004_sub</th>\n",
       "      <th>Ga0485162_metabat2_ours.116</th>\n",
       "      <th>...</th>\n",
       "      <th>depth</th>\n",
       "      <th>wtemp_in_celsius</th>\n",
       "      <th>DO_sat_perc</th>\n",
       "      <th>DO_mg_L</th>\n",
       "      <th>specific_conductivity</th>\n",
       "      <th>PH</th>\n",
       "      <th>chlorophyll_RFU</th>\n",
       "      <th>phycocyanin_RFU</th>\n",
       "      <th>Dissolved_organic_matter_RFU</th>\n",
       "      <th>Turbidity_RFU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ME_2020_07_24_15m_B.fastq Read Count</td>\n",
       "      <td>929</td>\n",
       "      <td>136</td>\n",
       "      <td>21252</td>\n",
       "      <td>78567</td>\n",
       "      <td>203159</td>\n",
       "      <td>1456</td>\n",
       "      <td>296</td>\n",
       "      <td>285</td>\n",
       "      <td>1456</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>12.03</td>\n",
       "      <td>-1.7</td>\n",
       "      <td>-0.18</td>\n",
       "      <td>581.66</td>\n",
       "      <td>7.37</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.53</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ME_2020_07_24_23_5M_B.fastq Read Count</td>\n",
       "      <td>12747</td>\n",
       "      <td>264</td>\n",
       "      <td>7253</td>\n",
       "      <td>11930</td>\n",
       "      <td>83501</td>\n",
       "      <td>585</td>\n",
       "      <td>49</td>\n",
       "      <td>80</td>\n",
       "      <td>4612</td>\n",
       "      <td>...</td>\n",
       "      <td>23.5</td>\n",
       "      <td>10.31</td>\n",
       "      <td>-1.5</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>600.14</td>\n",
       "      <td>7.17</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.88</td>\n",
       "      <td>12.65</td>\n",
       "      <td>1.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ME_2020_07_24_5m_C.fastq Read Count</td>\n",
       "      <td>24</td>\n",
       "      <td>538</td>\n",
       "      <td>7615</td>\n",
       "      <td>201</td>\n",
       "      <td>50</td>\n",
       "      <td>8188</td>\n",
       "      <td>139</td>\n",
       "      <td>401</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>25.05</td>\n",
       "      <td>109.1</td>\n",
       "      <td>9.00</td>\n",
       "      <td>492.70</td>\n",
       "      <td>8.62</td>\n",
       "      <td>0.56</td>\n",
       "      <td>1.70</td>\n",
       "      <td>9.70</td>\n",
       "      <td>4.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ME_2020_08_05_10M_B.fastq Read Count</td>\n",
       "      <td>40</td>\n",
       "      <td>589</td>\n",
       "      <td>3019</td>\n",
       "      <td>305572</td>\n",
       "      <td>3168</td>\n",
       "      <td>5636</td>\n",
       "      <td>155</td>\n",
       "      <td>1471</td>\n",
       "      <td>49</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>18.50</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>-0.07</td>\n",
       "      <td>570.32</td>\n",
       "      <td>7.47</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.87</td>\n",
       "      <td>11.63</td>\n",
       "      <td>0.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ME_2020_08_05_15M_B.fastq Read Count</td>\n",
       "      <td>416</td>\n",
       "      <td>199</td>\n",
       "      <td>15025</td>\n",
       "      <td>60601</td>\n",
       "      <td>381617</td>\n",
       "      <td>1039</td>\n",
       "      <td>127</td>\n",
       "      <td>289</td>\n",
       "      <td>1084</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11.76</td>\n",
       "      <td>-1.1</td>\n",
       "      <td>-0.12</td>\n",
       "      <td>582.32</td>\n",
       "      <td>7.41</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.67</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Unnamed: 0  Ga0485165_metabat1.097_sub  \\\n",
       "0    ME_2020_07_24_15m_B.fastq Read Count                         929   \n",
       "1  ME_2020_07_24_23_5M_B.fastq Read Count                       12747   \n",
       "2     ME_2020_07_24_5m_C.fastq Read Count                          24   \n",
       "3    ME_2020_08_05_10M_B.fastq Read Count                          40   \n",
       "4    ME_2020_08_05_15M_B.fastq Read Count                         416   \n",
       "\n",
       "   Ga0485163_maxbin.015_sub  Ga0485161_metabat1.109_sub  \\\n",
       "0                       136                       21252   \n",
       "1                       264                        7253   \n",
       "2                       538                        7615   \n",
       "3                       589                        3019   \n",
       "4                       199                       15025   \n",
       "\n",
       "   Ga0485163_metabat1.131  Ga0485165_metabat2_ours.031  \\\n",
       "0                   78567                       203159   \n",
       "1                   11930                        83501   \n",
       "2                     201                           50   \n",
       "3                  305572                         3168   \n",
       "4                   60601                       381617   \n",
       "\n",
       "   Ga0485163_metabat1.023_sub  Ga0485171_metabat2_ours.194  \\\n",
       "0                        1456                          296   \n",
       "1                         585                           49   \n",
       "2                        8188                          139   \n",
       "3                        5636                          155   \n",
       "4                        1039                          127   \n",
       "\n",
       "   Ga0485163_maxbin.004_sub  Ga0485162_metabat2_ours.116  ...  depth  \\\n",
       "0                       285                         1456  ...   15.0   \n",
       "1                        80                         4612  ...   23.5   \n",
       "2                       401                           16  ...    5.0   \n",
       "3                      1471                           49  ...   10.0   \n",
       "4                       289                         1084  ...   15.0   \n",
       "\n",
       "   wtemp_in_celsius  DO_sat_perc  DO_mg_L  specific_conductivity    PH  \\\n",
       "0             12.03         -1.7    -0.18                 581.66  7.37   \n",
       "1             10.31         -1.5    -0.16                 600.14  7.17   \n",
       "2             25.05        109.1     9.00                 492.70  8.62   \n",
       "3             18.50         -0.7    -0.07                 570.32  7.47   \n",
       "4             11.76         -1.1    -0.12                 582.32  7.41   \n",
       "\n",
       "   chlorophyll_RFU  phycocyanin_RFU  Dissolved_organic_matter_RFU  \\\n",
       "0             0.14             0.53                         12.18   \n",
       "1             0.35             0.88                         12.65   \n",
       "2             0.56             1.70                          9.70   \n",
       "3             0.33             0.87                         11.63   \n",
       "4             0.22             0.67                         12.18   \n",
       "\n",
       "   Turbidity_RFU  \n",
       "0           0.59  \n",
       "1           1.71  \n",
       "2           4.99  \n",
       "3           0.86  \n",
       "4           0.72  \n",
       "\n",
       "[5 rows x 442 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_to_phylum_map = dict(zip(taxonomy_df['Genome'], taxonomy_df['Phylum']))\n",
    "\n",
    "# List of genome names to map to Phylum\n",
    "genomes_list = [\n",
    "    'Ga0485157_metabat1.059',\n",
    "    'Ga0485167_maxbin.109',\n",
    "    'Ga0485162_maxbin.089',\n",
    "    'Ga0485161_maxbin.110',\n",
    "    'Ga0485161_maxbin.075',\n",
    "    'Ga0485157_metabat1.036',\n",
    "    'Ga0485165_metabat2_ours.012_sub',\n",
    "    'Ga0485169_maxbin.201_sub',\n",
    "    'Ga0485172_maxbin.081_sub',\n",
    "    'Ga0485168_maxbin.153'\n",
    "    \n",
    "]\n",
    "# Resetting the DataFrame to its original state before reapplying changes with the new requirement\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "\n",
    "# Applying the updated mapping with sequence numbers\n",
    "for i, genome in enumerate(genomes_list, start=1):\n",
    "    phylum = genome_to_phylum_map.get(genome)\n",
    "    if phylum and genome in t_count_df.columns:\n",
    "        new_column_name = f\"{phylum}_{i}\"  # Append underscore and sequence number\n",
    "        t_count_df = t_count_df.rename(columns={genome: new_column_name})\n",
    "\n",
    "# Display the first few rows of the modified DataFrame to verify changes\n",
    "t_count_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "89979c23",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "directory = \"metatransciptome\"\n",
    "\n",
    "# Check if the directory exists\n",
    "if not os.path.exists(directory):\n",
    "    os.makedirs(directory)  # Create the directory if it does not exist\n",
    "\n",
    "# Path for saving the CSV file\n",
    "file_path = os.path.join(directory, 'tax_10.csv')\n",
    "\n",
    "# Save the DataFrame to CSV\n",
    "t_count_df.to_csv(file_path, index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d428f239",
   "metadata": {},
   "source": [
    "# Top 15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "086bb1c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Ga0485165_metabat1.097_sub</th>\n",
       "      <th>Ga0485163_maxbin.015_sub</th>\n",
       "      <th>Ga0485161_metabat1.109_sub</th>\n",
       "      <th>Ga0485163_metabat1.131</th>\n",
       "      <th>Ga0485165_metabat2_ours.031</th>\n",
       "      <th>Ga0485163_metabat1.023_sub</th>\n",
       "      <th>Ga0485171_metabat2_ours.194</th>\n",
       "      <th>Ga0485163_maxbin.004_sub</th>\n",
       "      <th>Ga0485162_metabat2_ours.116</th>\n",
       "      <th>...</th>\n",
       "      <th>depth</th>\n",
       "      <th>wtemp_in_celsius</th>\n",
       "      <th>DO_sat_perc</th>\n",
       "      <th>DO_mg_L</th>\n",
       "      <th>specific_conductivity</th>\n",
       "      <th>PH</th>\n",
       "      <th>chlorophyll_RFU</th>\n",
       "      <th>phycocyanin_RFU</th>\n",
       "      <th>Dissolved_organic_matter_RFU</th>\n",
       "      <th>Turbidity_RFU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ME_2020_07_24_15m_B.fastq Read Count</td>\n",
       "      <td>929</td>\n",
       "      <td>136</td>\n",
       "      <td>21252</td>\n",
       "      <td>78567</td>\n",
       "      <td>203159</td>\n",
       "      <td>1456</td>\n",
       "      <td>296</td>\n",
       "      <td>285</td>\n",
       "      <td>1456</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>12.03</td>\n",
       "      <td>-1.7</td>\n",
       "      <td>-0.18</td>\n",
       "      <td>581.66</td>\n",
       "      <td>7.37</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.53</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ME_2020_07_24_23_5M_B.fastq Read Count</td>\n",
       "      <td>12747</td>\n",
       "      <td>264</td>\n",
       "      <td>7253</td>\n",
       "      <td>11930</td>\n",
       "      <td>83501</td>\n",
       "      <td>585</td>\n",
       "      <td>49</td>\n",
       "      <td>80</td>\n",
       "      <td>4612</td>\n",
       "      <td>...</td>\n",
       "      <td>23.5</td>\n",
       "      <td>10.31</td>\n",
       "      <td>-1.5</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>600.14</td>\n",
       "      <td>7.17</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.88</td>\n",
       "      <td>12.65</td>\n",
       "      <td>1.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ME_2020_07_24_5m_C.fastq Read Count</td>\n",
       "      <td>24</td>\n",
       "      <td>538</td>\n",
       "      <td>7615</td>\n",
       "      <td>201</td>\n",
       "      <td>50</td>\n",
       "      <td>8188</td>\n",
       "      <td>139</td>\n",
       "      <td>401</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>25.05</td>\n",
       "      <td>109.1</td>\n",
       "      <td>9.00</td>\n",
       "      <td>492.70</td>\n",
       "      <td>8.62</td>\n",
       "      <td>0.56</td>\n",
       "      <td>1.70</td>\n",
       "      <td>9.70</td>\n",
       "      <td>4.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ME_2020_08_05_10M_B.fastq Read Count</td>\n",
       "      <td>40</td>\n",
       "      <td>589</td>\n",
       "      <td>3019</td>\n",
       "      <td>305572</td>\n",
       "      <td>3168</td>\n",
       "      <td>5636</td>\n",
       "      <td>155</td>\n",
       "      <td>1471</td>\n",
       "      <td>49</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>18.50</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>-0.07</td>\n",
       "      <td>570.32</td>\n",
       "      <td>7.47</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.87</td>\n",
       "      <td>11.63</td>\n",
       "      <td>0.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ME_2020_08_05_15M_B.fastq Read Count</td>\n",
       "      <td>416</td>\n",
       "      <td>199</td>\n",
       "      <td>15025</td>\n",
       "      <td>60601</td>\n",
       "      <td>381617</td>\n",
       "      <td>1039</td>\n",
       "      <td>127</td>\n",
       "      <td>289</td>\n",
       "      <td>1084</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11.76</td>\n",
       "      <td>-1.1</td>\n",
       "      <td>-0.12</td>\n",
       "      <td>582.32</td>\n",
       "      <td>7.41</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.67</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Unnamed: 0  Ga0485165_metabat1.097_sub  \\\n",
       "0    ME_2020_07_24_15m_B.fastq Read Count                         929   \n",
       "1  ME_2020_07_24_23_5M_B.fastq Read Count                       12747   \n",
       "2     ME_2020_07_24_5m_C.fastq Read Count                          24   \n",
       "3    ME_2020_08_05_10M_B.fastq Read Count                          40   \n",
       "4    ME_2020_08_05_15M_B.fastq Read Count                         416   \n",
       "\n",
       "   Ga0485163_maxbin.015_sub  Ga0485161_metabat1.109_sub  \\\n",
       "0                       136                       21252   \n",
       "1                       264                        7253   \n",
       "2                       538                        7615   \n",
       "3                       589                        3019   \n",
       "4                       199                       15025   \n",
       "\n",
       "   Ga0485163_metabat1.131  Ga0485165_metabat2_ours.031  \\\n",
       "0                   78567                       203159   \n",
       "1                   11930                        83501   \n",
       "2                     201                           50   \n",
       "3                  305572                         3168   \n",
       "4                   60601                       381617   \n",
       "\n",
       "   Ga0485163_metabat1.023_sub  Ga0485171_metabat2_ours.194  \\\n",
       "0                        1456                          296   \n",
       "1                         585                           49   \n",
       "2                        8188                          139   \n",
       "3                        5636                          155   \n",
       "4                        1039                          127   \n",
       "\n",
       "   Ga0485163_maxbin.004_sub  Ga0485162_metabat2_ours.116  ...  depth  \\\n",
       "0                       285                         1456  ...   15.0   \n",
       "1                        80                         4612  ...   23.5   \n",
       "2                       401                           16  ...    5.0   \n",
       "3                      1471                           49  ...   10.0   \n",
       "4                       289                         1084  ...   15.0   \n",
       "\n",
       "   wtemp_in_celsius  DO_sat_perc  DO_mg_L  specific_conductivity    PH  \\\n",
       "0             12.03         -1.7    -0.18                 581.66  7.37   \n",
       "1             10.31         -1.5    -0.16                 600.14  7.17   \n",
       "2             25.05        109.1     9.00                 492.70  8.62   \n",
       "3             18.50         -0.7    -0.07                 570.32  7.47   \n",
       "4             11.76         -1.1    -0.12                 582.32  7.41   \n",
       "\n",
       "   chlorophyll_RFU  phycocyanin_RFU  Dissolved_organic_matter_RFU  \\\n",
       "0             0.14             0.53                         12.18   \n",
       "1             0.35             0.88                         12.65   \n",
       "2             0.56             1.70                          9.70   \n",
       "3             0.33             0.87                         11.63   \n",
       "4             0.22             0.67                         12.18   \n",
       "\n",
       "   Turbidity_RFU  \n",
       "0           0.59  \n",
       "1           1.71  \n",
       "2           4.99  \n",
       "3           0.86  \n",
       "4           0.72  \n",
       "\n",
       "[5 rows x 442 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# List of genome names to map to Phylum\n",
    "genomes_list = [\n",
    "    'Ga0485157_metabat1.059',\n",
    "    'Ga0485167_maxbin.109',\n",
    "    'Ga0485162_maxbin.089',\n",
    "    'Ga0485161_maxbin.110',\n",
    "    'Ga0485161_maxbin.075',\n",
    "    'Ga0485157_metabat1.036',\n",
    "    'Ga0485165_metabat2_ours.012_sub',\n",
    "    'Ga0485169_maxbin.201_sub',\n",
    "    'Ga0485172_maxbin.081_sub',\n",
    "    'Ga0485168_maxbin.153',\n",
    "    'Ga0485172_metabat2_ours.083',\n",
    "    'Ga0485162_maxbin.023',\n",
    "    'Ga0485160_maxbin.092',\n",
    "    'Ga0485158_metabat2_jgi.024',\n",
    "    'Ga0485162_metabat1.001'\n",
    "]\n",
    "\n",
    "\n",
    "# Applying the updated mapping with sequence numbers\n",
    "for i, genome in enumerate(genomes_list, start=1):\n",
    "    phylum = genome_to_phylum_map.get(genome)\n",
    "    if phylum and genome in t_count_df.columns:\n",
    "        new_column_name = f\"{phylum}_{i}\"  # Append underscore and sequence number\n",
    "        t_count_df = t_count_df.rename(columns={genome: new_column_name})\n",
    "\n",
    "# Display the first few rows of the modified DataFrame to verify changes\n",
    "t_count_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "526c3501",
   "metadata": {},
   "outputs": [],
   "source": [
    "t_count_df.to_csv('metatransciptome/tax_15.csv', index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f5a35e6",
   "metadata": {},
   "source": [
    "# betweenness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "a9fc0dc8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "p__Desulfobacterota_1\n",
      "p__Planctomycetota_2\n",
      "p__Proteobacteria_3\n",
      "p__Chloroflexota_4\n",
      "p__Bacteroidota_5\n",
      "p__Verrucomicrobiota_6\n",
      "p__Proteobacteria_7\n",
      "p__Actinobacteriota_8\n",
      "p__Krumholzibacteriota_9\n",
      "p__Armatimonadota_10\n",
      "p__Bacteroidota_11\n",
      "p__Verrucomicrobiota_12\n",
      "p__Actinobacteriota_13\n",
      "p__Firmicutes_A_14\n",
      "p__Desulfobacterota_15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Ga0485165_metabat1.097_sub</th>\n",
       "      <th>Ga0485163_maxbin.015_sub</th>\n",
       "      <th>Ga0485161_metabat1.109_sub</th>\n",
       "      <th>Ga0485163_metabat1.131</th>\n",
       "      <th>Ga0485165_metabat2_ours.031</th>\n",
       "      <th>Ga0485163_metabat1.023_sub</th>\n",
       "      <th>Ga0485171_metabat2_ours.194</th>\n",
       "      <th>Ga0485163_maxbin.004_sub</th>\n",
       "      <th>p__Verrucomicrobiota_6</th>\n",
       "      <th>...</th>\n",
       "      <th>depth</th>\n",
       "      <th>wtemp_in_celsius</th>\n",
       "      <th>DO_sat_perc</th>\n",
       "      <th>DO_mg_L</th>\n",
       "      <th>specific_conductivity</th>\n",
       "      <th>PH</th>\n",
       "      <th>chlorophyll_RFU</th>\n",
       "      <th>phycocyanin_RFU</th>\n",
       "      <th>Dissolved_organic_matter_RFU</th>\n",
       "      <th>Turbidity_RFU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ME_2020_07_24_15m_B.fastq Read Count</td>\n",
       "      <td>929</td>\n",
       "      <td>136</td>\n",
       "      <td>21252</td>\n",
       "      <td>78567</td>\n",
       "      <td>203159</td>\n",
       "      <td>1456</td>\n",
       "      <td>296</td>\n",
       "      <td>285</td>\n",
       "      <td>1456</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>12.03</td>\n",
       "      <td>-1.7</td>\n",
       "      <td>-0.18</td>\n",
       "      <td>581.66</td>\n",
       "      <td>7.37</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.53</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ME_2020_07_24_23_5M_B.fastq Read Count</td>\n",
       "      <td>12747</td>\n",
       "      <td>264</td>\n",
       "      <td>7253</td>\n",
       "      <td>11930</td>\n",
       "      <td>83501</td>\n",
       "      <td>585</td>\n",
       "      <td>49</td>\n",
       "      <td>80</td>\n",
       "      <td>4612</td>\n",
       "      <td>...</td>\n",
       "      <td>23.5</td>\n",
       "      <td>10.31</td>\n",
       "      <td>-1.5</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>600.14</td>\n",
       "      <td>7.17</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.88</td>\n",
       "      <td>12.65</td>\n",
       "      <td>1.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ME_2020_07_24_5m_C.fastq Read Count</td>\n",
       "      <td>24</td>\n",
       "      <td>538</td>\n",
       "      <td>7615</td>\n",
       "      <td>201</td>\n",
       "      <td>50</td>\n",
       "      <td>8188</td>\n",
       "      <td>139</td>\n",
       "      <td>401</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>25.05</td>\n",
       "      <td>109.1</td>\n",
       "      <td>9.00</td>\n",
       "      <td>492.70</td>\n",
       "      <td>8.62</td>\n",
       "      <td>0.56</td>\n",
       "      <td>1.70</td>\n",
       "      <td>9.70</td>\n",
       "      <td>4.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ME_2020_08_05_10M_B.fastq Read Count</td>\n",
       "      <td>40</td>\n",
       "      <td>589</td>\n",
       "      <td>3019</td>\n",
       "      <td>305572</td>\n",
       "      <td>3168</td>\n",
       "      <td>5636</td>\n",
       "      <td>155</td>\n",
       "      <td>1471</td>\n",
       "      <td>49</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>18.50</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>-0.07</td>\n",
       "      <td>570.32</td>\n",
       "      <td>7.47</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.87</td>\n",
       "      <td>11.63</td>\n",
       "      <td>0.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ME_2020_08_05_15M_B.fastq Read Count</td>\n",
       "      <td>416</td>\n",
       "      <td>199</td>\n",
       "      <td>15025</td>\n",
       "      <td>60601</td>\n",
       "      <td>381617</td>\n",
       "      <td>1039</td>\n",
       "      <td>127</td>\n",
       "      <td>289</td>\n",
       "      <td>1084</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11.76</td>\n",
       "      <td>-1.1</td>\n",
       "      <td>-0.12</td>\n",
       "      <td>582.32</td>\n",
       "      <td>7.41</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.67</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Unnamed: 0  Ga0485165_metabat1.097_sub  \\\n",
       "0    ME_2020_07_24_15m_B.fastq Read Count                         929   \n",
       "1  ME_2020_07_24_23_5M_B.fastq Read Count                       12747   \n",
       "2     ME_2020_07_24_5m_C.fastq Read Count                          24   \n",
       "3    ME_2020_08_05_10M_B.fastq Read Count                          40   \n",
       "4    ME_2020_08_05_15M_B.fastq Read Count                         416   \n",
       "\n",
       "   Ga0485163_maxbin.015_sub  Ga0485161_metabat1.109_sub  \\\n",
       "0                       136                       21252   \n",
       "1                       264                        7253   \n",
       "2                       538                        7615   \n",
       "3                       589                        3019   \n",
       "4                       199                       15025   \n",
       "\n",
       "   Ga0485163_metabat1.131  Ga0485165_metabat2_ours.031  \\\n",
       "0                   78567                       203159   \n",
       "1                   11930                        83501   \n",
       "2                     201                           50   \n",
       "3                  305572                         3168   \n",
       "4                   60601                       381617   \n",
       "\n",
       "   Ga0485163_metabat1.023_sub  Ga0485171_metabat2_ours.194  \\\n",
       "0                        1456                          296   \n",
       "1                         585                           49   \n",
       "2                        8188                          139   \n",
       "3                        5636                          155   \n",
       "4                        1039                          127   \n",
       "\n",
       "   Ga0485163_maxbin.004_sub  p__Verrucomicrobiota_6  ...  depth  \\\n",
       "0                       285                    1456  ...   15.0   \n",
       "1                        80                    4612  ...   23.5   \n",
       "2                       401                      16  ...    5.0   \n",
       "3                      1471                      49  ...   10.0   \n",
       "4                       289                    1084  ...   15.0   \n",
       "\n",
       "   wtemp_in_celsius  DO_sat_perc  DO_mg_L  specific_conductivity    PH  \\\n",
       "0             12.03         -1.7    -0.18                 581.66  7.37   \n",
       "1             10.31         -1.5    -0.16                 600.14  7.17   \n",
       "2             25.05        109.1     9.00                 492.70  8.62   \n",
       "3             18.50         -0.7    -0.07                 570.32  7.47   \n",
       "4             11.76         -1.1    -0.12                 582.32  7.41   \n",
       "\n",
       "   chlorophyll_RFU  phycocyanin_RFU  Dissolved_organic_matter_RFU  \\\n",
       "0             0.14             0.53                         12.18   \n",
       "1             0.35             0.88                         12.65   \n",
       "2             0.56             1.70                          9.70   \n",
       "3             0.33             0.87                         11.63   \n",
       "4             0.22             0.67                         12.18   \n",
       "\n",
       "   Turbidity_RFU  \n",
       "0           0.59  \n",
       "1           1.71  \n",
       "2           4.99  \n",
       "3           0.86  \n",
       "4           0.72  \n",
       "\n",
       "[5 rows x 442 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of genome names to map to Phylum\n",
    "genomes_list = [\n",
    "    \"Ga0485158_maxbin.120_sub\",\n",
    "\"Ga0485170_maxbin.090\",\n",
    "\"Ga0485159_metabat1.040\",\n",
    "\"Ga0485167_metabat2_ours.163\",\n",
    "\"Ga0485166_metabat2_ours.038\",\n",
    "\"Ga0485162_metabat2_ours.116\",\n",
    "\"Ga0485162_metabat2_ours.050\",\n",
    "\"Ga0485163_metabat1.181\",\n",
    "\"Ga0485158_metabat2_ours.098\",\n",
    "\"Ga0485169_metabat2_ours.035\",\n",
    "\"Ga0485159_metabat2_ours.130_sub\",\n",
    "\"Ga0485163_metabat1.115_sub\",\n",
    "\"Ga0485159_maxbin.025\",\n",
    "\"Ga0485161_metabat1.096\",\n",
    "\"Ga0485167_metabat1.127_sub\"\n",
    "\n",
    "]\n",
    "# Load the taxonomy file again (repeating due to re-upload)\n",
    "taxonomy_file_path = 'Fw_ lake mendota data/MAG_taxonomy.tsv'\n",
    "taxonomy_df = pd.read_csv(taxonomy_file_path, sep='\\t')\n",
    "\n",
    "# Load the T-count file\n",
    "t_count_file_path = 'metatransciptome_count.csv'\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "genome_to_phylum_map = dict(zip(taxonomy_df['Genome'], taxonomy_df['Phylum']))\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "\n",
    "\n",
    "# Applying the updated mapping with sequence numbers\n",
    "for i, genome in enumerate(genomes_list, start=1):\n",
    "    phylum = genome_to_phylum_map.get(genome)\n",
    "    if phylum and genome in t_count_df.columns:\n",
    "        new_column_name = f\"{phylum}_{i}\"  # Append underscore and sequence number\n",
    "        print(new_column_name)\n",
    "        t_count_df = t_count_df.rename(columns={genome: new_column_name})\n",
    "\n",
    "# Display the first few rows of the modified DataFrame to verify changes\n",
    "t_count_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "d399fdc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "t_count_df.to_csv('metatransciptome/tax_Betweenness.csv', index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb58d38e",
   "metadata": {},
   "source": [
    "# Closeness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "9756021f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "p__Verrucomicrobiota_1\n",
      "p__Actinobacteriota_2\n",
      "p__Bacteroidota_3\n",
      "p__Bacteroidota_4\n",
      "p__Bacteroidota_5\n",
      "p__Actinobacteriota_6\n",
      "p__Bacteroidota_7\n",
      "p__Bacteroidota_8\n",
      "p__Actinobacteriota_9\n",
      "p__Proteobacteria_10\n",
      "p__Planctomycetota_11\n",
      "p__Proteobacteria_12\n",
      "p__Proteobacteria_13\n",
      "p__Bacteroidota_14\n",
      "p__Verrucomicrobiota_15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Ga0485165_metabat1.097_sub</th>\n",
       "      <th>Ga0485163_maxbin.015_sub</th>\n",
       "      <th>p__Actinobacteriota_9</th>\n",
       "      <th>Ga0485163_metabat1.131</th>\n",
       "      <th>Ga0485165_metabat2_ours.031</th>\n",
       "      <th>Ga0485163_metabat1.023_sub</th>\n",
       "      <th>Ga0485171_metabat2_ours.194</th>\n",
       "      <th>Ga0485163_maxbin.004_sub</th>\n",
       "      <th>Ga0485162_metabat2_ours.116</th>\n",
       "      <th>...</th>\n",
       "      <th>depth</th>\n",
       "      <th>wtemp_in_celsius</th>\n",
       "      <th>DO_sat_perc</th>\n",
       "      <th>DO_mg_L</th>\n",
       "      <th>specific_conductivity</th>\n",
       "      <th>PH</th>\n",
       "      <th>chlorophyll_RFU</th>\n",
       "      <th>phycocyanin_RFU</th>\n",
       "      <th>Dissolved_organic_matter_RFU</th>\n",
       "      <th>Turbidity_RFU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ME_2020_07_24_15m_B.fastq Read Count</td>\n",
       "      <td>929</td>\n",
       "      <td>136</td>\n",
       "      <td>21252</td>\n",
       "      <td>78567</td>\n",
       "      <td>203159</td>\n",
       "      <td>1456</td>\n",
       "      <td>296</td>\n",
       "      <td>285</td>\n",
       "      <td>1456</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>12.03</td>\n",
       "      <td>-1.7</td>\n",
       "      <td>-0.18</td>\n",
       "      <td>581.66</td>\n",
       "      <td>7.37</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.53</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ME_2020_07_24_23_5M_B.fastq Read Count</td>\n",
       "      <td>12747</td>\n",
       "      <td>264</td>\n",
       "      <td>7253</td>\n",
       "      <td>11930</td>\n",
       "      <td>83501</td>\n",
       "      <td>585</td>\n",
       "      <td>49</td>\n",
       "      <td>80</td>\n",
       "      <td>4612</td>\n",
       "      <td>...</td>\n",
       "      <td>23.5</td>\n",
       "      <td>10.31</td>\n",
       "      <td>-1.5</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>600.14</td>\n",
       "      <td>7.17</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.88</td>\n",
       "      <td>12.65</td>\n",
       "      <td>1.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ME_2020_07_24_5m_C.fastq Read Count</td>\n",
       "      <td>24</td>\n",
       "      <td>538</td>\n",
       "      <td>7615</td>\n",
       "      <td>201</td>\n",
       "      <td>50</td>\n",
       "      <td>8188</td>\n",
       "      <td>139</td>\n",
       "      <td>401</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>25.05</td>\n",
       "      <td>109.1</td>\n",
       "      <td>9.00</td>\n",
       "      <td>492.70</td>\n",
       "      <td>8.62</td>\n",
       "      <td>0.56</td>\n",
       "      <td>1.70</td>\n",
       "      <td>9.70</td>\n",
       "      <td>4.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ME_2020_08_05_10M_B.fastq Read Count</td>\n",
       "      <td>40</td>\n",
       "      <td>589</td>\n",
       "      <td>3019</td>\n",
       "      <td>305572</td>\n",
       "      <td>3168</td>\n",
       "      <td>5636</td>\n",
       "      <td>155</td>\n",
       "      <td>1471</td>\n",
       "      <td>49</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>18.50</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>-0.07</td>\n",
       "      <td>570.32</td>\n",
       "      <td>7.47</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.87</td>\n",
       "      <td>11.63</td>\n",
       "      <td>0.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ME_2020_08_05_15M_B.fastq Read Count</td>\n",
       "      <td>416</td>\n",
       "      <td>199</td>\n",
       "      <td>15025</td>\n",
       "      <td>60601</td>\n",
       "      <td>381617</td>\n",
       "      <td>1039</td>\n",
       "      <td>127</td>\n",
       "      <td>289</td>\n",
       "      <td>1084</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11.76</td>\n",
       "      <td>-1.1</td>\n",
       "      <td>-0.12</td>\n",
       "      <td>582.32</td>\n",
       "      <td>7.41</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.67</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Unnamed: 0  Ga0485165_metabat1.097_sub  \\\n",
       "0    ME_2020_07_24_15m_B.fastq Read Count                         929   \n",
       "1  ME_2020_07_24_23_5M_B.fastq Read Count                       12747   \n",
       "2     ME_2020_07_24_5m_C.fastq Read Count                          24   \n",
       "3    ME_2020_08_05_10M_B.fastq Read Count                          40   \n",
       "4    ME_2020_08_05_15M_B.fastq Read Count                         416   \n",
       "\n",
       "   Ga0485163_maxbin.015_sub  p__Actinobacteriota_9  Ga0485163_metabat1.131  \\\n",
       "0                       136                  21252                   78567   \n",
       "1                       264                   7253                   11930   \n",
       "2                       538                   7615                     201   \n",
       "3                       589                   3019                  305572   \n",
       "4                       199                  15025                   60601   \n",
       "\n",
       "   Ga0485165_metabat2_ours.031  Ga0485163_metabat1.023_sub  \\\n",
       "0                       203159                        1456   \n",
       "1                        83501                         585   \n",
       "2                           50                        8188   \n",
       "3                         3168                        5636   \n",
       "4                       381617                        1039   \n",
       "\n",
       "   Ga0485171_metabat2_ours.194  Ga0485163_maxbin.004_sub  \\\n",
       "0                          296                       285   \n",
       "1                           49                        80   \n",
       "2                          139                       401   \n",
       "3                          155                      1471   \n",
       "4                          127                       289   \n",
       "\n",
       "   Ga0485162_metabat2_ours.116  ...  depth  wtemp_in_celsius  DO_sat_perc  \\\n",
       "0                         1456  ...   15.0             12.03         -1.7   \n",
       "1                         4612  ...   23.5             10.31         -1.5   \n",
       "2                           16  ...    5.0             25.05        109.1   \n",
       "3                           49  ...   10.0             18.50         -0.7   \n",
       "4                         1084  ...   15.0             11.76         -1.1   \n",
       "\n",
       "   DO_mg_L  specific_conductivity    PH  chlorophyll_RFU  phycocyanin_RFU  \\\n",
       "0    -0.18                 581.66  7.37             0.14             0.53   \n",
       "1    -0.16                 600.14  7.17             0.35             0.88   \n",
       "2     9.00                 492.70  8.62             0.56             1.70   \n",
       "3    -0.07                 570.32  7.47             0.33             0.87   \n",
       "4    -0.12                 582.32  7.41             0.22             0.67   \n",
       "\n",
       "   Dissolved_organic_matter_RFU  Turbidity_RFU  \n",
       "0                         12.18           0.59  \n",
       "1                         12.65           1.71  \n",
       "2                          9.70           4.99  \n",
       "3                         11.63           0.86  \n",
       "4                         12.18           0.72  \n",
       "\n",
       "[5 rows x 442 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of genome names to map to Phylum\n",
    "genomes_list = [\n",
    "    \"Ga0485157_metabat1.069\",\n",
    "\"Ga0485157_metabat2_ours.025\",\n",
    "\"Ga0485157_metabat2_ours.091\",\n",
    "\"Ga0485158_metabat2_ours.184\",\n",
    "\"Ga0485159_maxbin.015\",\n",
    "\"Ga0485160_maxbin.155_sub\",\n",
    "\"Ga0485160_metabat2_ours.051\",\n",
    "\"Ga0485160_metabat2_ours.071\",\n",
    "\"Ga0485161_metabat1.109_sub\",\n",
    "\"Ga0485162_metabat2_ours.036\",\n",
    "\"Ga0485163_maxbin.002_sub\",\n",
    "\"Ga0485163_maxbin.104_sub\",\n",
    "\"Ga0485163_maxbin.129\",\n",
    "\"Ga0485163_maxbin.130_sub\",\n",
    "\"Ga0485163_maxbin.183_sub\",\n",
    "\n",
    "]\n",
    "# Load the taxonomy file again (repeating due to re-upload)\n",
    "taxonomy_file_path = 'Fw_ lake mendota data/MAG_taxonomy.tsv'\n",
    "taxonomy_df = pd.read_csv(taxonomy_file_path, sep='\\t')\n",
    "\n",
    "# Load the T-count file\n",
    "t_count_file_path = 'metatransciptome_count.csv'\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "genome_to_phylum_map = dict(zip(taxonomy_df['Genome'], taxonomy_df['Phylum']))\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "\n",
    "\n",
    "# Applying the updated mapping with sequence numbers\n",
    "for i, genome in enumerate(genomes_list, start=1):\n",
    "    phylum = genome_to_phylum_map.get(genome)\n",
    "    if phylum and genome in t_count_df.columns:\n",
    "        new_column_name = f\"{phylum}_{i}\"  # Append underscore and sequence number\n",
    "        print(new_column_name)\n",
    "        t_count_df = t_count_df.rename(columns={genome: new_column_name})\n",
    "\n",
    "# Display the first few rows of the modified DataFrame to verify changes\n",
    "t_count_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "059a3b22",
   "metadata": {},
   "outputs": [],
   "source": [
    "t_count_df.to_csv('metatransciptome/tax_Closeness.csv', index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5ab8223",
   "metadata": {},
   "source": [
    "# Page Rank"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "9cb46ead",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "p__Krumholzibacteriota_1\n",
      "p__Bacteroidota_2\n",
      "p__Verrucomicrobiota_3\n",
      "p__Planctomycetota_4\n",
      "p__Verrucomicrobiota_5\n",
      "p__Planctomycetota_6\n",
      "p__Proteobacteria_7\n",
      "p__Proteobacteria_8\n",
      "p__Bacteroidota_9\n",
      "p__Bacteroidota_10\n",
      "p__Bacteroidota_11\n",
      "p__Actinobacteriota_12\n",
      "p__Verrucomicrobiota_13\n",
      "p__Proteobacteria_14\n",
      "p__Planctomycetota_15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Ga0485165_metabat1.097_sub</th>\n",
       "      <th>Ga0485163_maxbin.015_sub</th>\n",
       "      <th>Ga0485161_metabat1.109_sub</th>\n",
       "      <th>Ga0485163_metabat1.131</th>\n",
       "      <th>Ga0485165_metabat2_ours.031</th>\n",
       "      <th>Ga0485163_metabat1.023_sub</th>\n",
       "      <th>Ga0485171_metabat2_ours.194</th>\n",
       "      <th>Ga0485163_maxbin.004_sub</th>\n",
       "      <th>Ga0485162_metabat2_ours.116</th>\n",
       "      <th>...</th>\n",
       "      <th>depth</th>\n",
       "      <th>wtemp_in_celsius</th>\n",
       "      <th>DO_sat_perc</th>\n",
       "      <th>DO_mg_L</th>\n",
       "      <th>specific_conductivity</th>\n",
       "      <th>PH</th>\n",
       "      <th>chlorophyll_RFU</th>\n",
       "      <th>phycocyanin_RFU</th>\n",
       "      <th>Dissolved_organic_matter_RFU</th>\n",
       "      <th>Turbidity_RFU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ME_2020_07_24_15m_B.fastq Read Count</td>\n",
       "      <td>929</td>\n",
       "      <td>136</td>\n",
       "      <td>21252</td>\n",
       "      <td>78567</td>\n",
       "      <td>203159</td>\n",
       "      <td>1456</td>\n",
       "      <td>296</td>\n",
       "      <td>285</td>\n",
       "      <td>1456</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>12.03</td>\n",
       "      <td>-1.7</td>\n",
       "      <td>-0.18</td>\n",
       "      <td>581.66</td>\n",
       "      <td>7.37</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.53</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ME_2020_07_24_23_5M_B.fastq Read Count</td>\n",
       "      <td>12747</td>\n",
       "      <td>264</td>\n",
       "      <td>7253</td>\n",
       "      <td>11930</td>\n",
       "      <td>83501</td>\n",
       "      <td>585</td>\n",
       "      <td>49</td>\n",
       "      <td>80</td>\n",
       "      <td>4612</td>\n",
       "      <td>...</td>\n",
       "      <td>23.5</td>\n",
       "      <td>10.31</td>\n",
       "      <td>-1.5</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>600.14</td>\n",
       "      <td>7.17</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.88</td>\n",
       "      <td>12.65</td>\n",
       "      <td>1.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ME_2020_07_24_5m_C.fastq Read Count</td>\n",
       "      <td>24</td>\n",
       "      <td>538</td>\n",
       "      <td>7615</td>\n",
       "      <td>201</td>\n",
       "      <td>50</td>\n",
       "      <td>8188</td>\n",
       "      <td>139</td>\n",
       "      <td>401</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>25.05</td>\n",
       "      <td>109.1</td>\n",
       "      <td>9.00</td>\n",
       "      <td>492.70</td>\n",
       "      <td>8.62</td>\n",
       "      <td>0.56</td>\n",
       "      <td>1.70</td>\n",
       "      <td>9.70</td>\n",
       "      <td>4.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ME_2020_08_05_10M_B.fastq Read Count</td>\n",
       "      <td>40</td>\n",
       "      <td>589</td>\n",
       "      <td>3019</td>\n",
       "      <td>305572</td>\n",
       "      <td>3168</td>\n",
       "      <td>5636</td>\n",
       "      <td>155</td>\n",
       "      <td>1471</td>\n",
       "      <td>49</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>18.50</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>-0.07</td>\n",
       "      <td>570.32</td>\n",
       "      <td>7.47</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.87</td>\n",
       "      <td>11.63</td>\n",
       "      <td>0.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ME_2020_08_05_15M_B.fastq Read Count</td>\n",
       "      <td>416</td>\n",
       "      <td>199</td>\n",
       "      <td>15025</td>\n",
       "      <td>60601</td>\n",
       "      <td>381617</td>\n",
       "      <td>1039</td>\n",
       "      <td>127</td>\n",
       "      <td>289</td>\n",
       "      <td>1084</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11.76</td>\n",
       "      <td>-1.1</td>\n",
       "      <td>-0.12</td>\n",
       "      <td>582.32</td>\n",
       "      <td>7.41</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.67</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Unnamed: 0  Ga0485165_metabat1.097_sub  \\\n",
       "0    ME_2020_07_24_15m_B.fastq Read Count                         929   \n",
       "1  ME_2020_07_24_23_5M_B.fastq Read Count                       12747   \n",
       "2     ME_2020_07_24_5m_C.fastq Read Count                          24   \n",
       "3    ME_2020_08_05_10M_B.fastq Read Count                          40   \n",
       "4    ME_2020_08_05_15M_B.fastq Read Count                         416   \n",
       "\n",
       "   Ga0485163_maxbin.015_sub  Ga0485161_metabat1.109_sub  \\\n",
       "0                       136                       21252   \n",
       "1                       264                        7253   \n",
       "2                       538                        7615   \n",
       "3                       589                        3019   \n",
       "4                       199                       15025   \n",
       "\n",
       "   Ga0485163_metabat1.131  Ga0485165_metabat2_ours.031  \\\n",
       "0                   78567                       203159   \n",
       "1                   11930                        83501   \n",
       "2                     201                           50   \n",
       "3                  305572                         3168   \n",
       "4                   60601                       381617   \n",
       "\n",
       "   Ga0485163_metabat1.023_sub  Ga0485171_metabat2_ours.194  \\\n",
       "0                        1456                          296   \n",
       "1                         585                           49   \n",
       "2                        8188                          139   \n",
       "3                        5636                          155   \n",
       "4                        1039                          127   \n",
       "\n",
       "   Ga0485163_maxbin.004_sub  Ga0485162_metabat2_ours.116  ...  depth  \\\n",
       "0                       285                         1456  ...   15.0   \n",
       "1                        80                         4612  ...   23.5   \n",
       "2                       401                           16  ...    5.0   \n",
       "3                      1471                           49  ...   10.0   \n",
       "4                       289                         1084  ...   15.0   \n",
       "\n",
       "   wtemp_in_celsius  DO_sat_perc  DO_mg_L  specific_conductivity    PH  \\\n",
       "0             12.03         -1.7    -0.18                 581.66  7.37   \n",
       "1             10.31         -1.5    -0.16                 600.14  7.17   \n",
       "2             25.05        109.1     9.00                 492.70  8.62   \n",
       "3             18.50         -0.7    -0.07                 570.32  7.47   \n",
       "4             11.76         -1.1    -0.12                 582.32  7.41   \n",
       "\n",
       "   chlorophyll_RFU  phycocyanin_RFU  Dissolved_organic_matter_RFU  \\\n",
       "0             0.14             0.53                         12.18   \n",
       "1             0.35             0.88                         12.65   \n",
       "2             0.56             1.70                          9.70   \n",
       "3             0.33             0.87                         11.63   \n",
       "4             0.22             0.67                         12.18   \n",
       "\n",
       "   Turbidity_RFU  \n",
       "0           0.59  \n",
       "1           1.71  \n",
       "2           4.99  \n",
       "3           0.86  \n",
       "4           0.72  \n",
       "\n",
       "[5 rows x 442 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of genome names to map to Phylum\n",
    "genomes_list = [\n",
    "    \"Ga0485158_metabat2_ours.098\",\n",
    "\"Ga0485166_metabat2_ours.038\",\n",
    "\"Ga0485167_metabat2_ours.023\",\n",
    "\"Ga0485170_maxbin.090\",\n",
    "\"Ga0485161_metabat2_ours.167_sub\",\n",
    "\"Ga0485171_maxbin.130_sub\",\n",
    "\"Ga0485161_maxbin.110\",\n",
    "\"Ga0485164_metabat2_ours.069_sub\",\n",
    "\"Ga0485171_metabat1.063\",\n",
    "\"Ga0485159_metabat2_ours.079\",\n",
    "\"Ga0485171_metabat2_ours.127_sub\",\n",
    "\"Ga0485159_metabat2_ours.155_sub\",\n",
    "\"Ga0485158_metabat1.076\",\n",
    "\"Ga0485171_metabat1.030\",\n",
    "\"Ga0485168_metabat2_ours.135_sub\"\n",
    "]\n",
    "# Load the taxonomy file again (repeating due to re-upload)\n",
    "taxonomy_file_path = 'Fw_ lake mendota data/MAG_taxonomy.tsv'\n",
    "taxonomy_df = pd.read_csv(taxonomy_file_path, sep='\\t')\n",
    "\n",
    "# Load the T-count file\n",
    "t_count_file_path = 'metatransciptome_count.csv'\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "genome_to_phylum_map = dict(zip(taxonomy_df['Genome'], taxonomy_df['Phylum']))\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "\n",
    "\n",
    "# Applying the updated mapping with sequence numbers\n",
    "for i, genome in enumerate(genomes_list, start=1):\n",
    "    phylum = genome_to_phylum_map.get(genome)\n",
    "    if phylum and genome in t_count_df.columns:\n",
    "        new_column_name = f\"{phylum}_{i}\"  # Append underscore and sequence number\n",
    "        print(new_column_name)\n",
    "        t_count_df = t_count_df.rename(columns={genome: new_column_name})\n",
    "\n",
    "# Display the first few rows of the modified DataFrame to verify changes\n",
    "t_count_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "65867506",
   "metadata": {},
   "outputs": [],
   "source": [
    "t_count_df.to_csv('metatransciptome/tax_Page_Rank.csv', index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7fae3f29",
   "metadata": {},
   "source": [
    "# Degree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "4dc065d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "p__Krumholzibacteriota_1\n",
      "p__Bacteroidota_2\n",
      "p__Bacteroidota_3\n",
      "p__Proteobacteria_4\n",
      "p__Planctomycetota_5\n",
      "p__Bacteroidota_6\n",
      "p__Chloroflexota_7\n",
      "p__Bacteroidota_8\n",
      "p__Verrucomicrobiota_9\n",
      "p__Actinobacteriota_10\n",
      "p__Planctomycetota_11\n",
      "p__Chloroflexota_12\n",
      "p__Proteobacteria_13\n",
      "p__Firmicutes_A_14\n",
      "p__Verrucomicrobiota_15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Ga0485165_metabat1.097_sub</th>\n",
       "      <th>Ga0485163_maxbin.015_sub</th>\n",
       "      <th>Ga0485161_metabat1.109_sub</th>\n",
       "      <th>Ga0485163_metabat1.131</th>\n",
       "      <th>Ga0485165_metabat2_ours.031</th>\n",
       "      <th>Ga0485163_metabat1.023_sub</th>\n",
       "      <th>Ga0485171_metabat2_ours.194</th>\n",
       "      <th>Ga0485163_maxbin.004_sub</th>\n",
       "      <th>Ga0485162_metabat2_ours.116</th>\n",
       "      <th>...</th>\n",
       "      <th>depth</th>\n",
       "      <th>wtemp_in_celsius</th>\n",
       "      <th>DO_sat_perc</th>\n",
       "      <th>DO_mg_L</th>\n",
       "      <th>specific_conductivity</th>\n",
       "      <th>PH</th>\n",
       "      <th>chlorophyll_RFU</th>\n",
       "      <th>phycocyanin_RFU</th>\n",
       "      <th>Dissolved_organic_matter_RFU</th>\n",
       "      <th>Turbidity_RFU</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ME_2020_07_24_15m_B.fastq Read Count</td>\n",
       "      <td>929</td>\n",
       "      <td>136</td>\n",
       "      <td>21252</td>\n",
       "      <td>78567</td>\n",
       "      <td>203159</td>\n",
       "      <td>1456</td>\n",
       "      <td>296</td>\n",
       "      <td>285</td>\n",
       "      <td>1456</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>12.03</td>\n",
       "      <td>-1.7</td>\n",
       "      <td>-0.18</td>\n",
       "      <td>581.66</td>\n",
       "      <td>7.37</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.53</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ME_2020_07_24_23_5M_B.fastq Read Count</td>\n",
       "      <td>12747</td>\n",
       "      <td>264</td>\n",
       "      <td>7253</td>\n",
       "      <td>11930</td>\n",
       "      <td>83501</td>\n",
       "      <td>585</td>\n",
       "      <td>49</td>\n",
       "      <td>80</td>\n",
       "      <td>4612</td>\n",
       "      <td>...</td>\n",
       "      <td>23.5</td>\n",
       "      <td>10.31</td>\n",
       "      <td>-1.5</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>600.14</td>\n",
       "      <td>7.17</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.88</td>\n",
       "      <td>12.65</td>\n",
       "      <td>1.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ME_2020_07_24_5m_C.fastq Read Count</td>\n",
       "      <td>24</td>\n",
       "      <td>538</td>\n",
       "      <td>7615</td>\n",
       "      <td>201</td>\n",
       "      <td>50</td>\n",
       "      <td>8188</td>\n",
       "      <td>139</td>\n",
       "      <td>401</td>\n",
       "      <td>16</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>25.05</td>\n",
       "      <td>109.1</td>\n",
       "      <td>9.00</td>\n",
       "      <td>492.70</td>\n",
       "      <td>8.62</td>\n",
       "      <td>0.56</td>\n",
       "      <td>1.70</td>\n",
       "      <td>9.70</td>\n",
       "      <td>4.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ME_2020_08_05_10M_B.fastq Read Count</td>\n",
       "      <td>40</td>\n",
       "      <td>589</td>\n",
       "      <td>3019</td>\n",
       "      <td>305572</td>\n",
       "      <td>3168</td>\n",
       "      <td>5636</td>\n",
       "      <td>155</td>\n",
       "      <td>1471</td>\n",
       "      <td>49</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>18.50</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>-0.07</td>\n",
       "      <td>570.32</td>\n",
       "      <td>7.47</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.87</td>\n",
       "      <td>11.63</td>\n",
       "      <td>0.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ME_2020_08_05_15M_B.fastq Read Count</td>\n",
       "      <td>416</td>\n",
       "      <td>199</td>\n",
       "      <td>15025</td>\n",
       "      <td>60601</td>\n",
       "      <td>381617</td>\n",
       "      <td>1039</td>\n",
       "      <td>127</td>\n",
       "      <td>289</td>\n",
       "      <td>1084</td>\n",
       "      <td>...</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11.76</td>\n",
       "      <td>-1.1</td>\n",
       "      <td>-0.12</td>\n",
       "      <td>582.32</td>\n",
       "      <td>7.41</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.67</td>\n",
       "      <td>12.18</td>\n",
       "      <td>0.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 442 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Unnamed: 0  Ga0485165_metabat1.097_sub  \\\n",
       "0    ME_2020_07_24_15m_B.fastq Read Count                         929   \n",
       "1  ME_2020_07_24_23_5M_B.fastq Read Count                       12747   \n",
       "2     ME_2020_07_24_5m_C.fastq Read Count                          24   \n",
       "3    ME_2020_08_05_10M_B.fastq Read Count                          40   \n",
       "4    ME_2020_08_05_15M_B.fastq Read Count                         416   \n",
       "\n",
       "   Ga0485163_maxbin.015_sub  Ga0485161_metabat1.109_sub  \\\n",
       "0                       136                       21252   \n",
       "1                       264                        7253   \n",
       "2                       538                        7615   \n",
       "3                       589                        3019   \n",
       "4                       199                       15025   \n",
       "\n",
       "   Ga0485163_metabat1.131  Ga0485165_metabat2_ours.031  \\\n",
       "0                   78567                       203159   \n",
       "1                   11930                        83501   \n",
       "2                     201                           50   \n",
       "3                  305572                         3168   \n",
       "4                   60601                       381617   \n",
       "\n",
       "   Ga0485163_metabat1.023_sub  Ga0485171_metabat2_ours.194  \\\n",
       "0                        1456                          296   \n",
       "1                         585                           49   \n",
       "2                        8188                          139   \n",
       "3                        5636                          155   \n",
       "4                        1039                          127   \n",
       "\n",
       "   Ga0485163_maxbin.004_sub  Ga0485162_metabat2_ours.116  ...  depth  \\\n",
       "0                       285                         1456  ...   15.0   \n",
       "1                        80                         4612  ...   23.5   \n",
       "2                       401                           16  ...    5.0   \n",
       "3                      1471                           49  ...   10.0   \n",
       "4                       289                         1084  ...   15.0   \n",
       "\n",
       "   wtemp_in_celsius  DO_sat_perc  DO_mg_L  specific_conductivity    PH  \\\n",
       "0             12.03         -1.7    -0.18                 581.66  7.37   \n",
       "1             10.31         -1.5    -0.16                 600.14  7.17   \n",
       "2             25.05        109.1     9.00                 492.70  8.62   \n",
       "3             18.50         -0.7    -0.07                 570.32  7.47   \n",
       "4             11.76         -1.1    -0.12                 582.32  7.41   \n",
       "\n",
       "   chlorophyll_RFU  phycocyanin_RFU  Dissolved_organic_matter_RFU  \\\n",
       "0             0.14             0.53                         12.18   \n",
       "1             0.35             0.88                         12.65   \n",
       "2             0.56             1.70                          9.70   \n",
       "3             0.33             0.87                         11.63   \n",
       "4             0.22             0.67                         12.18   \n",
       "\n",
       "   Turbidity_RFU  \n",
       "0           0.59  \n",
       "1           1.71  \n",
       "2           4.99  \n",
       "3           0.86  \n",
       "4           0.72  \n",
       "\n",
       "[5 rows x 442 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of genome names to map to Phylum\n",
    "genomes_list = [\n",
    "    \"Ga0485158_metabat2_ours.098\",\n",
    "\"Ga0485159_metabat2_ours.079\",\n",
    "\"Ga0485171_metabat2_ours.127_sub\",\n",
    "\"Ga0485171_metabat2_ours.004\",\n",
    "\"Ga0485170_maxbin.090\",\n",
    "\"Ga0485171_metabat1.063\",\n",
    "\"Ga0485157_metabat2_ours.019\",\n",
    "\"Ga0485166_metabat2_ours.038\",\n",
    "\"Ga0485167_metabat2_ours.023\",\n",
    "\"Ga0485170_maxbin.059_sub\",\n",
    "\"Ga0485171_maxbin.130_sub\",\n",
    "\"Ga0485160_metabat2_ours.158_sub\",\n",
    "\"Ga0485161_maxbin.110\",\n",
    "\"Ga0485161_metabat1.096\",\n",
    "\"Ga0485161_metabat2_ours.167_sub\"\n",
    "\n",
    "]\n",
    "# Load the taxonomy file again (repeating due to re-upload)\n",
    "taxonomy_file_path = 'Fw_ lake mendota data/MAG_taxonomy.tsv'\n",
    "taxonomy_df = pd.read_csv(taxonomy_file_path, sep='\\t')\n",
    "\n",
    "# Load the T-count file\n",
    "t_count_file_path = 'metatransciptome_count.csv'\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "genome_to_phylum_map = dict(zip(taxonomy_df['Genome'], taxonomy_df['Phylum']))\n",
    "t_count_df = pd.read_csv(t_count_file_path)\n",
    "\n",
    "\n",
    "# Applying the updated mapping with sequence numbers\n",
    "for i, genome in enumerate(genomes_list, start=1):\n",
    "    phylum = genome_to_phylum_map.get(genome)\n",
    "    if phylum and genome in t_count_df.columns:\n",
    "        new_column_name = f\"{phylum}_{i}\"  # Append underscore and sequence number\n",
    "        print(new_column_name)\n",
    "        t_count_df = t_count_df.rename(columns={genome: new_column_name})\n",
    "\n",
    "# Display the first few rows of the modified DataFrame to verify changes\n",
    "t_count_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "03e2adf9",
   "metadata": {},
   "outputs": [],
   "source": [
    "t_count_df.to_csv('metatransciptome/tax_Degree.csv', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96d1d623",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
