# code

This document provides explanations of the Python and R scripts in this repository, along with a tutorial on how to reproduce the experiment results. The primary focus of these scripts is on processing lake microbiome data and visualizing the results.

## Scripts Overview

| Script File                          | Description                                                                                                           |
|--------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| **Metagenomic_Networkfile.py**       | Generates a directory containing files using the SPIEC-EASI algorithm as well as fraction method for metagenomic data analysis. Serves as input for `Metagenomic_Network.Rmd`. |
| **Metatransciptome_Networkfile.py**  | Similar function for metatranscriptomic data, generating input files for `Metatransciptomic_Network.Rmd`.             |
| **lake-microbiome-figures.py**       | Analyzes microbial abundance in varied environmental conditions and generates visualizations.                         |
| **Metagenomic_Network.Rmd**          | R Markdown file that explores connections between MAGs and environmental parameters that includes seven different feature selection methods, four permutation tests, and three percentage analyses. Outputs CarLasso models.         |
| **Metatransciptomic_Network.Rmd**    | Complements the Metagenomic analysis by focusing on metatranscriptomic data to produce comparative CarLasso models.  |
| **Car-figures.R**                    | Uses the CARlasso model to identify and visualize connections between MAGs and environmental parameters.               |


## Prerequisites

Ensure you have the following software installed before running the scripts:

- **Python**: Version 3.8 or higher recommended.
- **R**: Version 4.0 or higher recommended.

## Running the Scripts


### Execute the Code

Run the Python and R scripts by executing these commands in your terminal:

```bash
# Generate network files
python3 Metagenomic_Networkfile.py
python3 Metatransciptome_Networkfile.py

# Run R Markdown analyses
Rscript -e "rmarkdown::render('Metagenomic_Network.Rmd')"
Rscript -e "rmarkdown::render('Metatransciptomic_Network.Rmd')"

# Generate figures
python3 lake-microbiome-figures.py
Rscript Car-figures.R

```
For running these code, automatically create the following folder as input of algorithms.

| Folder Name                     | File Name                   | Description                                                                                      |
|---------------------------------|-----------------------------|--------------------------------------------------------------------------------------------------|
| **Metagenomic_Networkfile**     | `tax_10.csv`                | Contains basic taxonomic data with sample-specific metrics for metagenomic analysis.             |
|      | `tax_15.csv`                | Extended taxonomic data providing additional metrics for deeper analysis.                        |
|      | `tax_Betweenness.csv`       | Includes betweenness centrality metrics for taxa, highlighting nodes that bridge network clusters.|
|      | `tax_Closeness.csv`         | Features closeness centrality metrics for taxa, indicating the closeness to all other nodes.     |
|      | `tax_D-B-C-E-P.csv`         | Combined file with degree, betweenness, closeness, eigenvector, and PageRank centrality metrics. |
|     | `tax_Degree.csv`            | Contains degree centrality metrics, showing the number of direct connections each taxon has.     |
|      | `tax_Page_Rank.csv`         | Provides PageRank centrality scores, highlighting the influence of each taxon within the network.|
| **Metatransciptome_Networkfile**| `tax_10.csv`                | Contains basic taxonomic data with sample-specific metrics for metatranscriptomic analysis.      |
| | `tax_15.csv`                | Extended taxonomic data providing additional metrics for deeper transcriptomic analysis.         |
| | `tax_Betweenness.csv`       | Includes betweenness centrality metrics for taxa, highlighting crucial nodes in the transcript network. |
| | `tax_Closeness.csv`         | Features closeness centrality metrics for taxa, indicating their closeness to all other nodes in the network. |
|| `tax_Degree.csv`            | Contains degree centrality metrics, showing the number of direct connections each taxon has in the transcriptome. |
| | `tax_Page_Rank.csv`         | Provides PageRank centrality scores, highlighting the influence of each taxon within the transcriptomic network. |
| **OutputFigures**               | -                           | Contains all figures generated by the scripts, showcasing visual representations of data analyses conducted in the research paper. |




For running Metagenomic_Network.Rmd, Metatransciptomic_Network.Rmd, Car-figures.R, the following folder are required: Metagenomic_Networkfile, Metagenomic_Networkfile



